lib/openstack_api/openstack_query.py

```diff
@@ -372,3 +379,76 @@ def find_non_existant_object_projects(
                     else:
                         selected_projects.update({project_id: [object_id]})
         return selected_projects
+
+    # pylint:disable=too-many-arguments,too-many-locals
+    def email_users(
```

Comment:
So this method fails the "and or or" test
It performs the query *and* triggers the sending of emails.

Instead we should break it down into something like:
```
def notify_users(...):
    found_items = perform_query()
    prepared_emails = prepare_emails()
    send_emails(prepared_emails)
```

It might seem a touch overkill at the moment, but methods like this organically grow over time so it's better to trim them into separate responsibilities early on.